BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -0,0 +1,230 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveCore
+import BigNumber
+
+public class TransactionConfirmationStore: ObservableObject {
+  struct State {
+    var value: String = ""
+    var symbol: String = ""
+    var fiat: String = ""
+    var gasValue: String = ""
+    var gasSymbol: String = ""
+    var gasFiat: String = ""
+    var gasAssetRatio: Double = 0.0
+    var totalFiat: String = ""
+    var isBalanceSufficient: Bool = true
+  }
+  @Published var state: State = .init()
+  @Published var isLoading: Bool = false
+  
+  private var assetRatios: [String: Double] = [:]
+  
+  let numberFormatter = NumberFormatter().then {
+    $0.numberStyle = .currency
+    $0.currencyCode = "USD"
+  }
+  
+  private let assetRatioController: BraveWalletAssetRatioController
+  private let rpcController: BraveWalletEthJsonRpcController
+  private let txController: BraveWalletEthTxController
+  private let tokenRegistry: BraveWalletERCTokenRegistry
+  private var selectedChain: BraveWallet.EthereumChain = .init()
+  private var activeTransaction: BraveWallet.TransactionInfo?
+  
+  init(
+    assetRatioController: BraveWalletAssetRatioController,
+    rpcController: BraveWalletEthJsonRpcController,
+    txController: BraveWalletEthTxController,
+    tokenRegistry: BraveWalletERCTokenRegistry
+  ) {
+    self.assetRatioController = assetRatioController
+    self.rpcController = rpcController
+    self.txController = txController
+    self.tokenRegistry = tokenRegistry
+    
+    self.txController.add(self)
+  }
+  
+  func updateGasValue(for transaction: BraveWallet.TransactionInfo) {
+    let gasLimit = transaction.txData.baseData.gasLimit
+    let gasFormatter = WeiFormatter(decimalFormatStyle: .gasFee(limit: gasLimit.removingHexPrefix, radix: .hex))
+    let gasFee: String
+    if transaction.isEIP1559Transaction {
+      gasFee = transaction.txData.maxFeePerGas
+    } else {
+      gasFee = transaction.txData.baseData.gasPrice
+    }
+    self.state.gasValue = gasFormatter.decimalString(
+      for: gasFee.removingHexPrefix,
+         radix: .hex,
+         decimals: Int(selectedChain.decimals)
```

Comment:
indentation is weird here.

Reply:
I believe txData.maxFeePerGas is in GWei. do we need to give decimals: Int(selectedChain.decimals) which i think will make it in Wei?

Reply:
Yeah thats Xcode :( Can't do anything about it unfortunately

Reply:
I tried it with 9 decimals and got the wrong values so I guess it is stored in Wei, but displayed in Gwei